File: ZayShopUI/src/app/Model/CartSystem.ts
import { Color } from "./Color";
import { ProductSize } from "./ProductSize";
import { Report } from "./Report";
import { User } from "./SignUpUser";

export class CartSystem{

    public id:number;
    public name:string;
    public price:number;
    public image:string;
    public quantity:number;
    public report: Report = new Report();
    public productSize:ProductSize = new ProductSize();
    public colors:Color = new Color();
    public username: string;
    public applicationUser: User = new User();
}

File: Zay.Shop/ShopLibrary/CartSystem.cs
using IdentityProjectPractise.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ShopLibrary
{
    public class CartSystem
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public float Price { get; set; }

        public byte[] Image { get; set; }

        public int Quantity { get; set; }
        public virtual Report report { get; set; }

        public virtual ProductSize productSize { get; set; }

        public virtual Colors colors { get; set; }
        [ForeignKey("applicationUser")]
        public string Username { get; set; }
        public virtual ApplicationUser applicationUser { get; set; }
    }
}
